lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -50,29 +53,51 @@ sub new {
             die "$@ \n" if $@;
             $self->{db} = $backend_module->new();
         };
-        die "$@ \n" if $@;
-    }
+        if ($@) {
+            warn "Failed to initialize the database backend module: [$@] \n";
+            die "Failed to initialize the database backend module \n" if $@;
+        }
 
     return ( $self );
 }
 
+sub handle_exception {
+    my ( $method ) = @_;
+
+    warn "Unexpected error in the $method API call: [$@] \n";
+    die "Unexpected error in the $method API call \n" if $@;
+}
+
 $json_schemas{version_info} = joi->object->strict;
 sub version_info {
     my ( $self ) = @_;
 
     my %ver;
-    $ver{zonemaster_engine} = Zonemaster::Engine->VERSION;
-    $ver{zonemaster_backend} = Zonemaster::Backend->VERSION;
+    eval {
+        $ver{zonemaster_engine} = Zonemaster::Engine->VERSION;
+        $ver{zonemaster_backend} = Zonemaster::Backend->VERSION;
 
+    };
+    if ($@) {
+        handle_exception('version_info');
+    }
+    
     return \%ver;
 }
 
 $json_schemas{profile_names} = joi->object->strict;
 sub profile_names {
     my ($self) = @_;
 
-    my @profiles = Zonemaster::Backend::Config->load_config()->ListPublicProfiles();
+    my @profiles;
+    eval {
```

Comment:
I sure wish we had a more reasonable JSON-RPC library so we didn't have to jump through all these hoops...

Reply:
Is there an update required here?

Reply:
No, I was just whining. I probably should have kept it to myself.